### ISSUE TITLE
GlobalMutex initialization error when running on Mac

### ISSUE THREAD

**Issue by psh0078** [sillsdev/languageforge-lexbox#1006]
When starting local web app on a Mac, the follow errors is thrown:
```
[local-web-app-for-develop] Unhandled exception. SIL.PlatformUtilities.NativeException: An error with the number, 2, ocurred.
[local-web-app-for-develop]    at SIL.Threading.GlobalMutex.LinuxGlobalMutexAdapter.Init(Boolean initiallyOwned)
[local-web-app-for-develop]    at SIL.Threading.GlobalMutex.Initialize()
[local-web-app-for-develop]    at SIL.WritingSystems.Sldr.Initialize(Boolean offlineMode, String sldrCachePath, DateTime embeddedAllTagsTime)
[local-web-app-for-develop]    at SIL.WritingSystems.Sldr.Initialize(Boolean offlineMode, String sldrCachePath)
[local-web-app-for-develop]    at SIL.WritingSystems.Sldr.Initialize(Boolean offlineMode)
[local-web-app-for-develop]    at FwDataMiniLcmBridge.LcmUtils.ProjectLoader.Init() in /Users/kafka/Dev/languageforge-lexbox/backend/FwLite/FwDataMiniLcmBridge/LcmUtils/ProjectLoader.cs:line 31
[local-web-app-for-develop]    at LocalWebApp.LocalWebAppServer.SetupAppServer(String[] args, Action`1 configure) in /Users/kafka/Dev/languageforge-lexbox/backend/FwLite/LocalWebApp/LocalWebAppServer.cs:line 25
[local-web-app-for-develop]    at Program.<Main>$(String[] args) in /Users/kafka/Dev/languageforge-lexbox/backend/FwLite/LocalWebApp/Program.cs:line 4
[local-web-app-for-develop]    at Program.<Main>(String[] args)
[local-web-app-for-develop] dotnet watch ❌ Exited with error code 134
```

**Comment by hahn-kev** [comment]
I suspect that `SIL.Threading.GlobalMutex` does not support Mac. This would need to be fixed in libPalaso. 

That said the Mutext in to dotnet 5+ supports all linux and mac, so maybe we can simplify the GlobalMutex code in Palaso to fix this issue.

**Comment by rmunn** [comment]
I would take a look at the libpalaso code that detects whether it's running on Linux or Windows, and see if it's returning either a false positive or a false negative for the relevant check(s) on OS X. I remember seeing something like this before, and the relevant code in libpalaso was doing the wrong check. (I think it was looking for something Mono-specific to detect running on Linux, but that check was failing running under .NET Core).

**Comment by rmunn** [comment]
I checked. The check in the libpalaso GlobalMutex code is `if (Platform.IsLinux) _adapter = new LinuxGlobalMutexAdapter(name);` and the Platform.IsLinux check is done by calling `uname` and checking if the output is "Linux". (It expects the output "Darwin" for a Mac). Thing is, that check seems to be correct, so I don't know why the LinuxGlobalMutexAdapter class is being used. It should have fallen through to the ExplicitGlobalMutexAdapter code, which uses the .NET Mutex class.

@psh0078 , can you run `uname` on your Mac and post the output? Also the output of `uname -a` would be helpful as well.

**Comment by hahn-kev** [comment]
Actually I don't think `ExplicitGlobalMutexAdapter` exists in the version of SIL.Core we're using. That was just added back in June, so we're probably just using an old version that doesn't have this support. I've updated the SIL.Core version we're suing in #1007, hoping that fixes the issue. Thanks to the platform.bible team for fixing these things for us.

**Comment by hahn-kev** [comment]
closed by #1007
